### ISSUE TITLE
How do we create a tag?

### ISSUE THREAD

**Issue by ashlee3209** [MarquezProject/marquez#1017]
Hi, I am trying to tag a dataset or a filed. 

For example adding a tag called _"test"_ to dataset _"dummy-dataset"_: `curl --location --request POST '{{baseUrl}}/namespaces/test-namespace/datasets/dummy-dataset/tags/test'`. 

But it returns `{"code":404,"message":"Tag 'test' not found."}`.  Similar error for tagging a field.

If I try adding tags when creating the dataset, for example:

```
curl -X PUT {{baseUrl}}/namespaces/test-namespace/datasets/dummy-dataset2 \
  -H 'Content-Type: application/json' \
  -d '{ 
        "type": "DB_TABLE",
        "physicalName": "public.dummytable2",
        "sourceName": "test-source",
        "fields": [
          {"name": "a", "type": "INTEGER", "tags": [{"name": "SENSITIVE", "description": "sensitive data"}], "description": "col a"},
          {"name": "b", "type": "TIMESTAMP", "description": "col b"},
          {"name": "c", "type": "INTEGER", "description": "col c"},
          {"name": "d", "type": "INTEGER", "description": "col d"}
        ],
        "description": "Dummy dataset 2."
      }'
```

It will return `{"code":400,"message":"Unable to process JSON"}`.

If I remove the "tags" field in the "fields", the call works. And there is no creating/deleting tags apis in the https://marquezproject.github.io/marquez/openapi.html

Is that something still under development? Thank you!

**Comment by KevinMellott91** [comment]
I can't speak to your roadmap question, but I can point out that additional tags can be added at the time of setup by modifying the contents of the _marquez.<env>.yml_ file. For example, by default it only contains the following tags.

I know that doesn't fully answer your question, but hoping it helps in the short term!

```yaml
tags:
  - name: PII
    description: Personally identifiable information
  - name: SENSITIVE
    description: Contains sensitive information
```

**Comment by henneberger** [comment]
Ah! There seems to be a mismatch between the OpenAPI specification and the implementation. For the particular example, I do believe the openapi spec just diverged slightly from the intended behavior. The set of tags should be a set of strings. For example:
```bash
curl -X PUT {{baseUrl}}/namespaces/test-namespace/datasets/dummy-dataset2 \
  -H 'Content-Type: application/json' \
  -d '{ 
        "type": "DB_TABLE",
        "physicalName": "public.dummytable2",
        "sourceName": "test-source",
        "fields": [
          {"name": "a", "type": "INTEGER", "tags": ["SENSITIVE"], "description": "col a"},
          {"name": "b", "type": "TIMESTAMP", "description": "col b"},
          {"name": "c", "type": "INTEGER", "description": "col c"},
          {"name": "d", "type": "INTEGER", "description": "col d"}
        ],
        "description": "Dummy dataset 2."
      }'
```

Thanks for uncovering this, I'll commit a fix for the openapi specification asap. Also, since you cannot create tags via the tag resource, I'll change the api so any tags are allowable in either the dataset or the field.

**Comment by ashlee3209** [comment]
Hi, I am trying the same request as @henneberger posted above to create a new dataset with a "SENSITIVE" field. The dataset was successfully created but the tag is missing. I was using the helm installation with 0.12.2 image. Also I checked in the marquez-api pod, there is marquez.dev.yml which contains the tags like:

```
tags:
  - name: PII
    description: Personally identifiable information
  - name: SENSITIVE
    description: Contains sensitive information
```

**Comment by henneberger** [comment]
Ah, sorry about that. I have the endpoint returning the latest tags in PR #1162

**Comment by julienledem** [comment]
resolved by #1162
